### PR TITLE
MGMT-18490: Add shell completion to CLI RPM

### PR DESCRIPTION
**Pull request by vraiti**
Installs the BASH autocompletion script for Flightctl CLI via the RPM.

### CODE REVIEW

File: packaging/rpm/flightctl.spec [hunk 1]
```diff
@@ -47,6 +47,9 @@ install -m 0755 packaging/greenboot/flightctl-agent-running-check.sh %{buildroot
 cp bin/flightctl-agent %{buildroot}/usr/bin
 cp packaging/must-gather/flightctl-must-gather %{buildroot}/usr/bin
 cp packaging/systemd/flightctl-agent.service %{buildroot}/usr/lib/systemd/system
+if [[ -d /etc/bash_completion.d ]]; then
+	bin/flightctl completion > /etc/bash_completion.d/flightctl-completion.sh
```
Comment (fzdarsky) [inline packaging/rpm/flightctl.spec:51]: Is this missing the argument of which shell to create the shell completion for?

Comment (vraiti) [reply packaging/rpm/flightctl.spec]: Not necessarily, Cobra can automatically detect the shell.

**Review by fzdarsky — APPROVED** [review]

File: packaging/rpm/flightctl.spec [hunk 2]
```diff
@@ -47,6 +47,10 @@ install -m 0755 packaging/greenboot/flightctl-agent-running-check.sh %{buildroot
 cp bin/flightctl-agent %{buildroot}/usr/bin
 cp packaging/must-gather/flightctl-must-gather %{buildroot}/usr/bin
 cp packaging/systemd/flightctl-agent.service %{buildroot}/usr/lib/systemd/system
+if [[ -d /etc/bash_completion.d ]]; then
+	bin/flightctl completion > /tmp/flightctl-completion.sh
+	sudo mv /tmp/flightctl-completion.sh /etc/bash_completion.d
```
Comment (mangelajo) [inline packaging/rpm/flightctl.spec:52]: We should put the files in the %{buildroot} directory, which will be packaged later in the rpm; see the link from Fedora project to find the right places (we have macro directories for that)

Also bash completions for system installed packages should not go to `/etc/` (that is for user installed, manually managed files).

See the documentation here. Once you have the .sh file in tmp, follow the instructions for the install script and the %files section.

https://docs.fedoraproject.org/ca/packaging-guidelines/ShellCompletions/

File: packaging/rpm/flightctl.spec [hunk 3]
```diff
@@ -47,6 +47,10 @@ install -m 0755 packaging/greenboot/flightctl-agent-running-check.sh %{buildroot
...
+fi
 
 %files
 /usr/bin/flightctl
```
Comment (mangelajo) [inline packaging/rpm/flightctl.spec:56]: you will also need to add an entry here for the new packaged file.

**Review by mangelajo — CHANGES_REQUESTED** [review]

**Comment by mangelajo** [comment]
Just some comments, packaging needs some time, but fedora has excellent documentation for almost everything.

File: packaging/rpm/flightctl.spec [hunk 4]
```diff
@@ -47,9 +47,18 @@ install -m 0755 packaging/greenboot/flightctl-agent-running-check.sh %{buildroot
 cp bin/flightctl-agent %{buildroot}/usr/bin
 cp packaging/must-gather/flightctl-must-gather %{buildroot}/usr/bin
 cp packaging/systemd/flightctl-agent.service %{buildroot}/usr/lib/systemd/system
+bin/flightctl completion bash > flightctl-completion.bash
+install -Dpm 0644 flightctl-completion.bash -t %{buildroot}/%{_datadir}/bash-completion/completions/flightctl-completion.bash
```
Comment (mangelajo) [inline packaging/rpm/flightctl.spec:51]: nice!!! :)

Did you use _datadir directly and not %{bash_completions_dir} etc for any particular reason? (I am wondering if it's not supported in the centos/rhel macros yet, or just an error)

File: packaging/rpm/flightctl.spec [hunk 5]
```diff
@@ -47,9 +47,18 @@ install -m 0755 packaging/greenboot/flightctl-agent-running-check.sh %{buildroot
...
+bin/flightctl completion fish > flightctl-completion.fish
+install -Dpm 0644 flightctl-completion.fish -t %{buildroot}/%{_datadir}/fish/vendor_completions.d/flightctl-completion.fish
+bin/flightctl completion zsh > _flightctl-completion
+install -Dpm 0644 _flightctl-completion -t %{buildroot}/%{_datadir}/zsh/site-functions/_flightctl-completion
 
 %files
 /usr/bin/flightctl
+%{_datadir}/bash-completion/completions/flightctl-completion.bash
+%{_datadir}/fish/vendor_completions.d/flightctl-completion.fish
+%{_datadir}/zsh/site-functions/_flightctl-completion
```
Comment (mangelajo) [inline packaging/rpm/flightctl.spec:61]: same question here.

File: packaging/rpm/flightctl.spec (hunk 4 above)
Comment (vraiti) [reply packaging/rpm/flightctl.spec:51]: I suspect that's the case. %{SHELL_completions_dir}-type macros would not expand resulting in a file not found error.

**Review by mangelajo — APPROVED** [review]

**Pull request merged**